harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -28,56 +27,198 @@
+        self._start_from_batch = steps_completed
+        self._val_from_previous_run = self._core_context.train._get_last_validation()
+        self._step_zero_validation = step_zero_validation
+
+        # Training configs
+        self._latest_checkpoint = latest_checkpoint
+        self._average_training_metrics = average_training_metrics
+        self._test_mode = test_mode
+        self._searcher_metric_name = searcher_metric_name
+        self._ckpt_policy = checkpoint_policy
+        self._smaller_is_better = smaller_is_better
+        self._global_batch_size = self._context.get_global_batch_size()
+
         if torch.cuda.is_available():
-            self.prof._set_sync_device(self._sync_device)
-        self.callbacks = self.trial.build_callbacks()
-        for callback in self.callbacks.values():
+            self._prof._set_sync_device(self._sync_device)
+        self._callbacks = self._trial.build_callbacks()
+        for callback in self._callbacks.values():
             if util.is_overridden(callback.on_checkpoint_end, pytorch.PyTorchCallback):
                 warnings.warn(
                     "The on_checkpoint_end callback is deprecated, please use "
                     "on_checkpoint_write_end instead.",
                     FutureWarning,
                 )
 
-        if len(self.context.models) == 0:
+        if len(self._context.models) == 0:
             raise det.errors.InvalidExperimentException(
                 "Must have at least one model. "
                 "This might be caused by not wrapping your model with wrap_model().",
             )
-        if len(self.context.optimizers) == 0:
+        if len(self._context.optimizers) == 0:
             raise det.errors.InvalidExperimentException(
                 "Must have at least one optimizer. "
                 "This might be caused by not wrapping your optimizer with wrap_optimizer().",
             )
         self._check_evaluate_implementation()
 
-        self.wlsq = None  # type: Optional[layers.WorkloadSequencer]
-        if self.workloads is None:
-            self.workloads, self.wlsq = layers.make_compatibility_workloads(
-                self.context._core,
-                self.env,
-                self.context.get_global_batch_size(),
-            )
-
-        self.steps_completed = self.env.steps_completed
-
         # Currently only horovod and torch backends are supported for distributed training
-        if self.context.distributed.size > 1:
+        if self._context.distributed.size > 1:
             assert (
-                self.use_horovod or self.use_torch
+                self._use_horovod or self._use_torch
             ), "Must use horovod or torch for distributed training"
+        if context.distributed.size > 1 and not self._is_chief:
+            log_level = logging.DEBUG if debug else logging.WARNING
+            logging.getLogger().setLevel(log_level)
+        self._metric_writer = self._create_metric_writer()
```

Comment:
I don't think that log levels should be set inside of an object.  They should only be set by the entrypoint script I think.

Reply:
not in the trainer either lol

Reply:
this is in the harness now.